spec/policy/oidc_authentication/oidc_authentication_spec.lua

```diff
@@ -0,0 +1,185 @@
+local _M = require('apicast.policy.oidc_authentication')
+local JWT = require('resty.jwt')
+local rsa = require('fixtures.rsa')
+local oidc_discovery = require('resty.oidc.discovery')
+local OIDC = require('apicast.oauth.oidc')
+local test_backend_client = require 'resty.http_ng.backend.test'
+
+local access_token = setmetatable({
+  header = { typ = 'JWT', alg = 'RS256', kid = 'somekid' },
+  payload = {
+    iss = 'http://example.com/issuer',
+    sub = 'some',
+    aud = 'one',
+    exp = ngx.now() + 3600,
+  },
+}, { __tostring = function(jwt) return JWT:sign(rsa.private, jwt) end })
+
+describe('oidc_authentication policy', function()
+  describe('.new', function()
+    it('works without configuration', function()
+      assert(_M.new())
+    end)
+
+    it('accepts empty configuration', function()
+        assert(_M.new({ }))
+    end)
+
+    it('accepts issuer_endpoint configuration', function()
+      assert(_M.new({ issuer_endpoint = 'http://example.com/path' }))
+    end)
+
+    it('ignores invalid issuer_endpoint configuration', function()
+      assert(_M.new({ issuer_endpoint = 'http: example.com path' }))
+    end)
+
+    local test_backend
+
+    before_each(function()
+      test_backend = test_backend_client.new()
+    end)
+
+    after_each(function()
+      test_backend.verify_no_outstanding_expectations()
+    end)
+
+    it('uses discovery', function()
+      local issuer_endpoint = 'http://example.com'
+
+      stub(oidc_discovery, 'http_backend', test_backend)
+
+      test_backend
+              .expect{ url = issuer_endpoint .. "/.well-known/openid-configuration" }
+              .respond_with{ status = 200, headers = { content_type = 'application/json;charset=UTF-8' },
+                             body = [[
+            {
+              "issuer": "https://example.com/issuer",
+              "jwks_uri": "https://example.com/jwks",
+              "id_token_signing_alg_values_supported": [ "RS256" ]
+            }
+          ]] }
+
+      test_backend
+              .expect{ url = "https://example.com/jwks" }
+              .respond_with{
+        status = 200,
+        headers = { content_type = 'application/json' },
+        body =  [[ { "keys": [{
+                      "kid": "3g-I9PWt6NrznPLcbE4zZrakXar27FDKEpqRPlD2i2Y",
+                      "kty": "RSA",
+                      "n": "iqXwBiZgN2q1dCKU1P_vzyiGacdQhfqgxQST7GFlWU_PUljV9uHrLOadWadpxRAuskNpXWsrKoU_hDxtSpUIRJj6hL5YTlrvv-IbFwPNtD8LnOfKL043_ZdSOe3aT4R4NrBxUomndILUESlhqddylVMCGXQ81OB73muc9ovR68Ajzn8KzpU_qegh8iHwk-SQvJxIIvgNJCJTC6BWnwS9Bw2ns0fQOZZRjWFRVh8BjkVdqa4vCAb6zw8hpR1y9uSNG-fqUAPHy5IYQaD8k8QX0obxJ0fld61fH-Wr3ENpn9YZWYBcKvnwLm2bvxqmNVBzW4rhGEZb9mf-KrSagD5GUw",
+                      "e": "AQAB"
+                    }] } ]]
+      }
+
+      _M.new{ issuer_endpoint = issuer_endpoint }
+    end)
+  end)
+
+  describe(':rewrite', function()
+    before_each(function()
+      ngx.var = {}
+    end)
+
+    it('parses access token', function()
```

Comment:
I'd also mention that it stores the parsed token in the context.
I think this is important because other policies might want to check if there's already a parsed jwt in there.